Refactor navigation block to use generic classnames.

jasmussen opened the pull request:
## Description

This is a work in progress and a followup to #33918. The goal of this PR is to simplify the navigation block CSS to target menu items in a generic way, so as to avoid drift when new menu items are added.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/accessibility/ -->
- [x] I've tested my changes with keyboard and screen readers. <!-- Instructions: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/accessibility-testing.md -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://developer.wordpress.org/coding-standards/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/code/native-mobile.md -->

jasmussen committed:
Refactor navigation block to use generic classnames.

tellthemachines commented on packages/block-library/src/navigation/style.scss:
```diff
@@ -1,8 +1,12 @@
 // Navigation block and menu item styles.
-// This also styles navigation links inside the Page List block,
 // as that block is meant to behave as menu items when leveraged.
```
Shouldn't this line also be removed?

tellthemachines commented on packages/block-library/src/navigation/style.scss:
```diff
@@ -308,8 +301,9 @@
+	// @todo: check this one separately.
 	.submenu-container, // This target items created by the Page List block.
-	.wp-block-navigation__container .wp-block-navigation-link__container {
+	.wp-block-navigation__container .wp-block-navigation__submenu-container {
```
We should be able to remove `submenu-container` here, and use just `.wp-block-navigation__submenu-container` to target all submenus. I'm not sure why the `.wp-block-navigation__container` was added in the first place; it's not needed to target only the submenus, because the submenu classname does that already, so it may have been added to increase specificity. But if we do need additional specificity, we can just duplicate `.wp-block-navigation__submenu-container`.

tellthemachines commented on packages/block-library/src/navigation/style.scss:
```diff
@@ -171,13 +168,13 @@
+	&:focus-within > .wp-block-navigation__submenu-container {
 		visibility: visible;
 		opacity: 1;
 	}
```
We should now be able to remove the custom page list hover and focus-within rules altogether - the ones between lines 182 and 199.

tellthemachines reviewed (approved):
Looks good! Much nicer without all the extra classnames. I left a few minor comments, but this is already a huge improvement over the current state of things 😄

jasmussen commented:
Thanks so much for the review 👌 — I will address your feedback and return!

jasmussen committed:
Address feedback.

jasmussen commented:
Thanks again for the review, I've addressed your points by implementing your suggestions one by one and it appears to be working really well all over. I'd still appreciate a good thorough review on this one, here's some example content:

```
<!-- wp:paragraph -->
<p>Horizontal nav</p>
<!-- /wp:paragraph -->

<!-- wp:navigation {"orientation":"horizontal"} -->
<!-- wp:navigation-link {"label":"Home","type":"page","id":11,"url":"http://localhost:8888/","kind":"post-type","isTopLevelLink":true} /-->

<!-- wp:navigation-link {"label":"Journal","type":"page","id":13,"url":"http://localhost:8888/?page_id=13","kind":"post-type","isTopLevelLink":true} /-->

<!-- wp:navigation-link {"label":"Contact","type":"page","id":15,"url":"http://localhost:8888/?page_id=15","kind":"post-type","isTopLevelLink":true} /-->

<!-- wp:navigation-link {"label":"About","type":"page","id":648,"url":"http://localhost:8888/?page_id=648","kind":"post-type","isTopLevelLink":true} -->
<!-- wp:navigation-link {"label":"Services","type":"page","id":656,"url":"http://localhost:8888/?page_id=656","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"Properties","type":"page","id":657,"url":"http://localhost:8888/?page_id=657","kind":"post-type","isTopLevelLink":false} -->
<!-- wp:navigation-link {"label":"Test","type":"page","id":2976,"url":"http://localhost:8888/?page_id=2976","kind":"post-type","isTopLevelLink":false} /-->
<!-- /wp:navigation-link -->

<!-- wp:navigation-link {"label":"Qualities","type":"page","id":658,"url":"http://localhost:8888/?page_id=658","kind":"post-type","isTopLevelLink":false} /-->
<!-- /wp:navigation-link -->

<!-- wp:navigation-link {"label":"Contact","type":"page","id":3056,"url":"http://localhost:8888/?page_id=3056","kind":"post-type","isTopLevelLink":true} /-->

<!-- wp:navigation-link {"label":"Directions","type":"page","id":6,"url":"http://localhost:8888/?page_id=6","kind":"post-type","isTopLevelLink":true} /-->

<!-- wp:navigation-link {"label":"Make a donation","type":"page","id":3057,"url":"http://localhost:8888/?page_id=3057","kind":"post-type","isTopLevelLink":true} /-->

<!-- wp:navigation-link {"label":"About","type":"page","id":3055,"url":"http://localhost:8888/?page_id=3055","kind":"post-type","isTopLevelLink":true} -->
<!-- wp:navigation-link {"label":"History","type":"page","id":45,"url":"http://localhost:8888/?page_id=45","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"Faculty","type":"page","id":57,"url":"http://localhost:8888/?page_id=57","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"Distinguished Alumni","type":"page","id":59,"url":"http://localhost:8888/?page_id=59","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"Diversity and Inclusion","type":"page","id":61,"url":"http://localhost:8888/?page_id=61","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"Leadership","type":"page","id":63,"url":"http://localhost:8888/?page_id=63","kind":"post-type","isTopLevelLink":false} /-->
<!-- /wp:navigation-link -->

<!-- wp:navigation-link {"label":"Admissions","type":"page","id":3058,"url":"http://localhost:8888/?page_id=3058","kind":"post-type","isTopLevelLink":true} -->
<!-- wp:navigation-link {"label":"Career Paths","type":"page","id":65,"url":"http://localhost:8888/?page_id=65","kind":"post-type","isTopLevelLink":false} -->
<!-- wp:navigation-link {"label":"Technology","type":"page","id":67,"url":"http://localhost:8888/?page_id=67","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"Business","type":"page","id":68,"url":"http://localhost:8888/?page_id=68","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"Design","type":"page","id":3062,"url":"http://localhost:8888/?page_id=3062","kind":"post-type","isTopLevelLink":false} /-->
<!-- /wp:navigation-link -->

<!-- wp:navigation-link {"label":"Scholarship \u0026#038; Financial Aid","type":"page","id":33,"url":"http://localhost:8888/?page_id=33","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"Tuition","type":"page","id":31,"url":"http://localhost:8888/?page_id=31","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"Programs","type":"page","id":12,"url":"http://localhost:8888/?page_id=12","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"Undergraduate Admissions","type":"page","id":3059,"url":"http://localhost:8888/?page_id=3059","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"Graduate Admissions","type":"page","id":3060,"url":"http://localhost:8888/?page_id=3060","kind":"post-type","isTopLevelLink":false} /-->
<!-- /wp:navigation-link -->

<!-- wp:navigation-link {"label":"Student Life","type":"page","id":14,"url":"http://localhost:8888/?page_id=14","kind":"post-type","isTopLevelLink":true} -->
<!-- wp:navigation-link {"label":"Services","type":"page","id":39,"url":"http://localhost:8888/?page_id=39","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"Athletics","type":"page","id":41,"url":"http://localhost:8888/?page_id=41","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"Study Abroad Opportunities","type":"page","id":43,"url":"http://localhost:8888/?page_id=43","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"FAQs","type":"page","id":47,"url":"http://localhost:8888/?page_id=47","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"Tutoring Services","type":"page","id":49,"url":"http://localhost:8888/?page_id=49","kind":"post-type","isTopLevelLink":false} /-->
<!-- /wp:navigation-link -->

<!-- wp:navigation-link {"label":"Research","type":"page","id":16,"url":"http://localhost:8888/?page_id=16","kind":"post-type","isTopLevelLink":true} -->
<!-- wp:navigation-link {"label":"Awards \u0026#038; Honors","type":"page","id":51,"url":"http://localhost:8888/?page_id=51","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"Undergraduate Research","type":"page","id":53,"url":"http://localhost:8888/?page_id=53","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"Graduate Research","type":"page","id":55,"url":"http://localhost:8888/?page_id=55","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"Partnerships","type":"page","id":35,"url":"http://localhost:8888/?page_id=35","kind":"post-type","isTopLevelLink":false} /-->
<!-- /wp:navigation-link -->

<!-- wp:navigation-link {"label":"News","type":"page","id":17,"url":"http://localhost:8888/?page_id=17","kind":"post-type","isTopLevelLink":true} /-->
<!-- /wp:navigation -->

<!-- wp:spacer -->
<div style="height:100px" aria-hidden="true" class="wp-block-spacer"></div>
<!-- /wp:spacer -->

<!-- wp:paragraph -->
<p>Horizontal page list</p>
<!-- /wp:paragraph -->

<!-- wp:navigation {"orientation":"horizontal"} -->
<!-- wp:page-list {"isNavigationChild":true} /-->
<!-- /wp:navigation -->

<!-- wp:spacer -->
<div style="height:100px" aria-hidden="true" class="wp-block-spacer"></div>
<!-- /wp:spacer -->

<!-- wp:paragraph -->
<p>Vertical page list</p>
<!-- /wp:paragraph -->

<!-- wp:navigation {"orientation":"vertical"} -->
<!-- wp:page-list {"isNavigationChild":true} /-->
<!-- /wp:navigation -->

<!-- wp:spacer -->
<div style="height:100px" aria-hidden="true" class="wp-block-spacer"></div>
<!-- /wp:spacer -->

<!-- wp:paragraph -->
<p>Vertical nav</p>
<!-- /wp:paragraph -->

<!-- wp:navigation {"orientation":"vertical"} -->
<!-- wp:navigation-link {"label":"Home","type":"page","id":11,"url":"http://localhost:8888/","kind":"post-type","isTopLevelLink":true} /-->

<!-- wp:navigation-link {"label":"Journal","type":"page","id":13,"url":"http://localhost:8888/?page_id=13","kind":"post-type","isTopLevelLink":true} /-->

<!-- wp:navigation-link {"label":"Contact","type":"page","id":15,"url":"http://localhost:8888/?page_id=15","kind":"post-type","isTopLevelLink":true} /-->

<!-- wp:navigation-link {"label":"About","type":"page","id":648,"url":"http://localhost:8888/?page_id=648","kind":"post-type","isTopLevelLink":true} -->
<!-- wp:navigation-link {"label":"Services","type":"page","id":656,"url":"http://localhost:8888/?page_id=656","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"Properties","type":"page","id":657,"url":"http://localhost:8888/?page_id=657","kind":"post-type","isTopLevelLink":false} -->
<!-- wp:navigation-link {"label":"Test","type":"page","id":2976,"url":"http://localhost:8888/?page_id=2976","kind":"post-type","isTopLevelLink":false} /-->
<!-- /wp:navigation-link -->

<!-- wp:navigation-link {"label":"Qualities","type":"page","id":658,"url":"http://localhost:8888/?page_id=658","kind":"post-type","isTopLevelLink":false} /-->
<!-- /wp:navigation-link -->

<!-- wp:navigation-link {"label":"About","type":"page","id":3055,"url":"http://localhost:8888/?page_id=3055","kind":"post-type","isTopLevelLink":true} -->
<!-- wp:navigation-link {"label":"History","type":"page","id":45,"url":"http://localhost:8888/?page_id=45","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"Faculty","type":"page","id":57,"url":"http://localhost:8888/?page_id=57","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"Distinguished Alumni","type":"page","id":59,"url":"http://localhost:8888/?page_id=59","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"Diversity and Inclusion","type":"page","id":61,"url":"http://localhost:8888/?page_id=61","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"Leadership","type":"page","id":63,"url":"http://localhost:8888/?page_id=63","kind":"post-type","isTopLevelLink":false} /-->
<!-- /wp:navigation-link -->

<!-- wp:navigation-link {"label":"Contact","type":"page","id":3056,"url":"http://localhost:8888/?page_id=3056","kind":"post-type","isTopLevelLink":true} /-->

<!-- wp:navigation-link {"label":"Directions","type":"page","id":6,"url":"http://localhost:8888/?page_id=6","kind":"post-type","isTopLevelLink":true} /-->

<!-- wp:navigation-link {"label":"Make a donation","type":"page","id":3057,"url":"http://localhost:8888/?page_id=3057","kind":"post-type","isTopLevelLink":true} /-->

<!-- wp:navigation-link {"label":"Admissions","type":"page","id":3058,"url":"http://localhost:8888/?page_id=3058","kind":"post-type","isTopLevelLink":true} -->
<!-- wp:navigation-link {"label":"Career Paths","type":"page","id":65,"url":"http://localhost:8888/?page_id=65","kind":"post-type","isTopLevelLink":false} -->
<!-- wp:navigation-link {"label":"Technology","type":"page","id":67,"url":"http://localhost:8888/?page_id=67","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"Business","type":"page","id":68,"url":"http://localhost:8888/?page_id=68","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"Design","type":"page","id":3062,"url":"http://localhost:8888/?page_id=3062","kind":"post-type","isTopLevelLink":false} /-->
<!-- /wp:navigation-link -->

<!-- wp:navigation-link {"label":"Scholarship \u0026#038; Financial Aid","type":"page","id":33,"url":"http://localhost:8888/?page_id=33","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"Tuition","type":"page","id":31,"url":"http://localhost:8888/?page_id=31","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"Programs","type":"page","id":12,"url":"http://localhost:8888/?page_id=12","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"Undergraduate Admissions","type":"page","id":3059,"url":"http://localhost:8888/?page_id=3059","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"Graduate Admissions","type":"page","id":3060,"url":"http://localhost:8888/?page_id=3060","kind":"post-type","isTopLevelLink":false} /-->
<!-- /wp:navigation-link -->

<!-- wp:navigation-link {"label":"Student Life","type":"page","id":14,"url":"http://localhost:8888/?page_id=14","kind":"post-type","isTopLevelLink":true} -->
<!-- wp:navigation-link {"label":"Services","type":"page","id":39,"url":"http://localhost:8888/?page_id=39","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"Athletics","type":"page","id":41,"url":"http://localhost:8888/?page_id=41","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"Study Abroad Opportunities","type":"page","id":43,"url":"http://localhost:8888/?page_id=43","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"FAQs","type":"page","id":47,"url":"http://localhost:8888/?page_id=47","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"Tutoring Services","type":"page","id":49,"url":"http://localhost:8888/?page_id=49","kind":"post-type","isTopLevelLink":false} /-->
<!-- /wp:navigation-link -->

<!-- wp:navigation-link {"label":"Research","type":"page","id":16,"url":"http://localhost:8888/?page_id=16","kind":"post-type","isTopLevelLink":true} -->
<!-- wp:navigation-link {"label":"Awards \u0026#038; Honors","type":"page","id":51,"url":"http://localhost:8888/?page_id=51","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"Undergraduate Research","type":"page","id":53,"url":"http://localhost:8888/?page_id=53","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"Graduate Research","type":"page","id":55,"url":"http://localhost:8888/?page_id=55","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"Partnerships","type":"page","id":35,"url":"http://localhost:8888/?page_id=35","kind":"post-type","isTopLevelLink":false} /-->
<!-- /wp:navigation-link -->

<!-- wp:navigation-link {"label":"News","type":"page","id":17,"url":"http://localhost:8888/?page_id=17","kind":"post-type","isTopLevelLink":true} /-->
<!-- /wp:navigation -->
```

One thing I noticed, the sub-submenu indicator in nav items look like this:

<img width="265" alt="Screenshot 2021-08-24 at 11 08 20" src="https://user-images.githubusercontent.com/1204802/130590102-e4af596e-a00e-4fc8-be9a-f5c36812f520.png">

The chevron is inside the a element:

<img width="690" alt="Screenshot 2021-08-24 at 11 08 00" src="https://user-images.githubusercontent.com/1204802/130590127-f5feb54f-f9d7-4ad9-8d7b-e7e682e55943.png">

The Page List looks like this:

<img width="282" alt="Screenshot 2021-08-24 at 11 08 23" src="https://user-images.githubusercontent.com/1204802/130590149-d6516ad0-91a5-48b4-9669-fb3226941043.png">

The chevron is outside of the a element:

<img width="717" alt="Screenshot 2021-08-24 at 11 08 12" src="https://user-images.githubusercontent.com/1204802/130590184-53824628-d7b5-42af-9f66-1ef5f70897e8.png">

I should be able to tweak that by editing the Page List block itself — any concerns with that?

As an additional question, now that we've refactored the nav block classnames, can we/should we retire or remove some of the existing CSS classes applied? Page list is probably a challenge since it's shipping, but what can or should we do for the nav block itself?

---

Note to self, as a followup I'd like to look at an issue where the submenus take up horizontal space even when they are invisible, in some cases causing horizontal scrollbars.

tellthemachines reviewed (approved):
This is looking good! 

Unfortunately the example markup doesn't work on my local because all my post and page URLs are different 😅 but I tested with both horizontal and vertical variations of Navigation, with multiple nested submenus on both Navigation items and Page list, so I think all the possibilities have been covered. 

(I did come across a couple of styling issues, but they're already on trunk so we can deal with them separately)

> I should be able to tweak that by editing the Page List block itself — any concerns with that?

It might be good to address that separately too, because it might conflict with the work we're doing in #33775 to enable open on click instead of focus. That PR doesn't touch Page List, but we will want Page List submenus to also open on click so they match the rest of the Nav, right?

> As an additional question, now that we've refactored the nav block classnames, can we/should we retire or remove some of the existing CSS classes applied? Page list is probably a challenge since it's shipping, but what can or should we do for the nav block itself?

Yes, we definitely should! And yes, we might want to leave Page List classes alone because the block is already in Core, but for the Navigation block and its specific children, we can remove the old classes. Again, best do a follow-up PR so we can merge this one already 😄

jasmussen commented:
Rebased, it seems like this one could be good to land soon!

> That PR doesn't touch Page List, but we will want Page List submenus to also open on click so they match the rest of the Nav, right?

Ideally the Page List would behave the same as any other menu items, yes. Speaking of, I've [explored some designs that leverage such a dropdown menu block](https://github.com/WordPress/gutenberg/issues/18395#issuecomment-905379168) but with a parent link inside, that considers the Page List as part of the equation.

mkaz commented:
The justify items do not seem to be working properly in the editor, but do appear to be in published view.
Both for horizontal and vertical navigation.

vcanales commented:
> I should be able to tweak that by editing the Page List block itself — any concerns with that?

Navigation links look like this on TT1; on your branch, and on `trunk`:

<img width="405" alt="image" src="https://user-images.githubusercontent.com/1157901/130826671-b134f640-bc94-4ad1-aad1-86255b7864fa.png">

I'm wondering if the arrow being stuck to the text is what's expected here, or if the separation would be better. If the former is true, then I think modifying the Page List block would be OK! Otherwise, we might win something by changing the navigation link markup instead, and helping TT1 to not look — in my humble opinion — this wonky.

tellthemachines commented:
> Navigation links look like this on TT1; on your branch, and on `trunk`:

Yeah, this is one of the issues I mentioned above. It's happening across the board, not only on TT1. It should also be mentioned that the formatting is different depending on whether the submenu is selected, or its parent.
With parent selected:
<img width="331" alt="Screen Shot 2021-08-26 at 11 27 03 am" src="https://user-images.githubusercontent.com/8096000/130885316-98302612-bba1-4975-82e5-821c750442be.png">
With submenu selected:
<img width="297" alt="Screen Shot 2021-08-26 at 11 27 11 am" src="https://user-images.githubusercontent.com/8096000/130885362-71e2b7fc-0dc2-4ea6-a530-a1fb7c859f49.png">

tellthemachines commented:
> The justify items do not seem to be working properly in the editor, but do appear to be in published view.
Both for horizontal and vertical navigation.

Oh, well spotted, @mkaz ! Justification isn't working at all for vertical nav on trunk either, so not a problem specific to this PR. For the horizontal nav though, it's the margins on the children that are being overridden by general block styles:
<img width="613" alt="  " src="https://user-images.githubusercontent.com/8096000/130900113-8a105381-d6a6-4c65-b4eb-71a1dd5d626f.png">
So I reckon we'll need to bump the specificity on `.wp-block-navigation__container>.wp-block-navigation-item, .wp-block-page-list, .wp-block-page-list>.wp-block-navigation-item` to fix it.

jasmussen commented:
> So I reckon we'll need to bump the specificity on .wp-block-navigation__container>.wp-block-navigation-item, .wp-block-page-list, .wp-block-page-list>.wp-block-navigation-item to fix it.

Great call and thanks for the advice.

I'm hesitant to do it in this PR, though, and in fact I'm hesitant to even do it at all. The rules are output by classic.scss file, i.e. the issue should not be present on block themes. In #32659 I tried my best to reduce the specificity of those classic styles so that we wouldn't have to increase specificity otherwise. The problem with increasing specificity is that we need low specificity for global styles/theme.json to be able to set the margins, as https://github.com/WordPress/gutenberg/pull/31878 sought to achieve. There's likely still a solution for classic themes, but I'd like to explore it separately.

In addition to this, TT1 has obsolete rules for styling the navigation block, which is causing those extra paddings. I've tracked that here: https://core.trac.wordpress.org/ticket/53220

So unless there are any objections, I think I'll land this in an hour or so, just to unblock other related PRs that rely on this. Let me know otherwise!

jasmussen commented:
Alright, landing this one and as always, happy to do followups.

jasmussen commented:
Should this one get a dev note? In case yes, here's a suggested note:

> In [34171](https://github.com/WordPress/gutenberg/pull/34171) the structural CSS for the navigation block was refactored to target new generic classsnames across menu items of multiple types including the Page List block. 
> The visual appearance should remain identical to before, but the changes may affect specificity in case your theme styles the navigation block beyond theme.json properties. Changed classes:
> - `.submenu-container` and `.wp-block-navigation-link__container` is now replaced by `.wp-block-navigation__submenu-container` (submenu containers)
> - `.wp-block-pages-list__item` and `.wp-block-navigation-link` is now replaced by `.wp-block-navigation-item` (menu items)
> - `.wp-block-navigation-link__content` is now replaced by `.wp-block-navigation-item__content` (the link inside a menu item)
> - `.wp-block-page-list__submenu-icon` and `.wp-block-navigation-link__submenu-icon` is now replaced by `.wp-block-navigation__submenu-icon` (dropdown chevron indicator)

tellthemachines commented:
It won't need an actual dev note on make/core because this block isn't in core yet, but it might be good to mention it in the plugin release notes.

talldan commented:
@jasmussen @tellthemachines Please can you make a point of testing the navigation editor with any block changes.

Things are looking a bit of a mess with all the changes to the block over the last couple of months.

talldan commented on packages/edit-navigation/src/components/editor/style.scss:
```diff
@@ -131,15 +131,15 @@
 
 		// When editing a link with children, highlight the parent
 		// and adjust the spacing and submenu icon.
 		.wp-block-navigation-link.has-child.is-editing {
```
Is `.wp-block-navigation-link` still a valid classname? I think this might be where the problem is. Should it be `wp-block-navigation-item`?

jasmussen replied:
It should be, yes. Fixed in #34344.

talldan commented on packages/base-styles/_z-index.scss:
```diff
@@ -43,8 +43,8 @@ $z-layers: (
 	// Navigation menu dropdown.
-	".has-child .wp-block-navigation-link__container": 28,
-	".has-child:hover .wp-block-navigation-link__container": 29,
+	".has-child .wp-block-navigation__submenu-container": 28,
```
It'd be great to have some docs in the navigation block's readme about how these classnames work.

jasmussen replied:
I took a stab at that here: https://github.com/WordPress/gutenberg/pull/34344/files#diff-b43a36c885bf6479506b3541da11760e584757fd78809954b3ff785f54a2e11dR47 — let me know if that looks right to you.

talldan commented:
Ok, I think I've managed to fix up many of the broken styles in the nav editor as part of https://github.com/WordPress/gutenberg/pull/34281

jasmussen commented:
Thanks for looking, Dan, sorry about the nav screen issues. I will create PRs for your points and let me know how I can help with 34281.

talldan commented:
@jasmussen No worries, no need to make PRs as the issues should be fixed in #34281. A review would be appreciated 😄 

I think we just need to make sure the nav editor undergoes a comparison with `trunk` with any changes to the block. Hopefully we can come up with a better solution so that isn't always required.

jasmussen commented:
I know you already have some of the navigation screen issues fixed, thanks for that. Nevertheless I created #34344 to fix a few additional issues, so some overlap did happen. However your #34281 still includes some fixes to the chevron open/close behavior that 34344 does not, so we'll eventually need both.
